marshal/claim.go

```diff
@@ -509,8 +541,20 @@ func (c *claim) healthCheck() bool {
 func (c *claim) healthCheckLoop() {
 	time.Sleep(<-c.marshal.cluster.jitters)
 	for !c.Terminated() {
-		// Update current offsets internally to the last processed before checking if we are healthy
-		c.updateOffsets()
+		// Attempt to update offsets; if this fails we want to to do a quicker retry
+		// than the jitter interval to allow us to try to retry some times before we
+		// give up.
+		for !c.heartbeatExpired() {
+			if err := c.updateOffsets(); err != nil {
```

Comment:
Correct. I think that's fine, if we can't talk to Kafka to get partition offsets then we've got some serious issues and I feel comfortable saying we can fail.